src/main/java/de/bonndan/nivio/observation/LandscapeObserverPool.java

```diff
@@ -0,0 +1,59 @@
+package de.bonndan.nivio.observation;
+
+import de.bonndan.nivio.model.Landscape;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.util.StringUtils;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * A wrapper around observers to reduce the async results to a single boolean.
+ *
+ *
+ */
+public class LandscapeObserverPool {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(LandscapeObserverPool.class);
+    public static final String DELIM = ";";
+
+    private final Landscape landscape;
+    private final List<URLObserver> observers;
+
+    public LandscapeObserverPool(Landscape landscape, List<URLObserver> observers) {
+        this.landscape = landscape;
+        this.observers = observers;
+    }
+
+    /**
+     * @return the change
+     */
+    public Optional<String> hasChange() {
+        LOGGER.info("Detecting changes in {} observers for landscape {}.", observers.size(), landscape.getIdentifier());
+
+        CompletableFuture<String>[] futures = observers.stream().map(URLObserver::hasChange).toArray(CompletableFuture[]::new);
+        CompletableFuture<Void> allDoneFuture = CompletableFuture.allOf(futures);
+
+        CompletableFuture<List<String>> listCompletableFuture = allDoneFuture.thenApply(
+                v -> Stream.of(futures)
+                        .map(CompletableFuture::join)
+                        .filter(Objects::nonNull)
+                        .collect(Collectors.toList())
+        );
+
+        List<String> changes = listCompletableFuture.join();
```

Comment:
Just for my understanding, since you have already done a join above in line 44, would a `.get()` also do the job?

Reply:
The human brain is only capable to understand how CompletableFutures work right in the moment when reading the manual. I don't know who wrote this code.

get() throws InterruptedException, join does not. Join would usually block, but there all fotures are already complete. I copied that from https://dzone.com/articles/java-8-completablefuture

Reply:
OK ;)